includes/sanitizers/class-amp-form-sanitizer.php

```diff
@@ -67,23 +67,41 @@ public function sanitize() {
 				$action_url = esc_url_raw( '//' . $_SERVER['HTTP_HOST'] . wp_unslash( $_SERVER['REQUEST_URI'] ) );
 			} else {
 				$action_url = $node->getAttribute( 'action' );
-
-				// Handle relative URLs.
-				if ( ! preg_match( '#^(https?:)?//#', $action_url ) ) {
-					$schemeless_host = '//' . $_SERVER['HTTP_HOST'];
-					if ( '?' === $action_url[0] || '#' === $action_url[0] ) {
-						// For actions consisting of only a query or URL fragment, include the schemeless-host and the REQUEST URI of the current page.
-						$action_url = $schemeless_host . wp_unslash( $_SERVER['REQUEST_URI'] ) . $action_url;
-					} elseif ( '.' === $action_url[0] ) {
-						// For actions consisting of relative paths (e.g. '../'), prepend the schemeless-host and a trailing-slashed REQUEST URI.
-						$action_url = $schemeless_host . trailingslashit( wp_unslash( $_SERVER['REQUEST_URI'] ) ) . $action_url;
-					} else {
-						// Otherwise, when the action URL includes an absolute path, just append it to the schemeless-host.
-						$action_url = $schemeless_host . $action_url;
+				$parsed_url = wp_parse_url( $action_url );
+
+				// Ignore a malformed URL - it will be later sanitized.
+				if ( false !== $parsed_url ) {
+					// Handle relative URLs.
+					if ( ! isset( $parsed_url['scheme'] ) || 'https' !== $parsed_url['scheme'] ) {
```

Comment:
Why the `'https' !== $parsed_url['scheme']` condition here?

Reply:
To ignore URLs with the `https` scheme, otherwise it becomes schemeless which is unnecessary.

I've updated the comment there to reflect this:
https://github.com/ampproject/amp-wp/blob/16b7fa8a7ee9b5ea7ebc47111f4495d8f6a54537/includes/sanitizers/class-amp-form-sanitizer.php#L132

Reply:
@pierlon I think I simplified this further, and also added short-circuiting when it is a scheme-less URL (in which case nothing is left to do either). Please see e99eade20214510dd9fbd34c3930116fa26c5fc7.

Reply:
Thanks! And its more readable :+1: .